### PR TITLE
Add experiments on fine-tuning

### PR DESCRIPTION
**Pull request by scott-sievert-3**
**What does this PR implement?**
It adds the framework for fine-tuning a small NN. It feeds images through a NN, saves the features/labels to disk, then reads those values in later for an experiment.

TODO:

- [x] pre-process dataset (write out embedding, ...)
- [x] get single model working
- [x] explore hyperparams (Adagrad, GeoDamp, SGD, AdaDamp, GD, ...)

**Reference issues/PRs**
This PR depends on https://github.com/stsievert/adadamp/pull/24

### CODE REVIEW

**Comment by stsievert** [comment]
Thank you for the contribution!
